projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/vdk.py

```diff
@@ -34,8 +34,11 @@ def run_job(path, arguments=None):
             )
             cmd = shlex.split(cmd_str)
             process = subprocess.Popen(
-                cmd, stdout=log_file, stderr=log_file, env=os.environ.copy()
+                cmd,
```

Comment:
I thought the [stackoverflow](https://stackoverflow.com/questions/21009416/python-subprocess-security) is clear but this one is clearer ? https://security.openstack.org/guidelines/dg_use-subprocess-securely.html

I think we can ignore the warning. But if you want to remove it. Most likely you need to split the "vdk" from the cmd_str string so that the analyzer realizes that the command (cmd[0]) is not user input . We likely can ignore the warning as the code is safe. But it's better not to have it as not to have doubt.

Reply:
Thank you, I understand it better now